src/ServiceResetError.c

```diff
@@ -84,7 +84,13 @@ void Ros_ServiceResetError_Trigger(const void* request_msg, void* response_msg)
     }
 
     // Check for condition that can be fixed remotely
-    if (Ros_Controller_IsAlarm())
+    if (Ros_Controller_IsMajorAlarm())
+    {
+        rosidl_runtime_c__String__assign(&response->message, "Major alarm active. Cannot be reset. Check teach pendant");
+        response->result_code.value = MOTION_NOT_READY_MAJOR_ALARM;
+        goto DONE;
+    } 
+    else if (Ros_Controller_IsAlarm())
```

Comment:
Because of the `goto DONE`, this is not needed.

I'd simplify by just making this an `if`.